server/resp/src/main/java/org/infinispan/server/resp/commands/json/JSONGET.java

```
@@ -0,0 +1,171 @@
+            .create(FunctionalMapImpl.create(handler.typedCache(null)));
+      int finalPos = args.pos();
+      CompletionStage<String> cs = cache.eval(key, view -> {
+         DefaultPrettyPrinter rpp = (args.space() != null) ? new RespPrettyPrinter(args.space()) : new RespPrettyPrinter();
+
+         Indenter ind = new DefaultIndenter(args.indent(), args.newline());
+         rpp.indentArraysWith(ind);
+         rpp.indentObjectsWith(ind);
+         int pathPos = finalPos;
+         ObjectMapper mapper = new ObjectMapper();
+         JsonDoc value = (JsonDoc) view.find().orElse(null);
+         var doc = value == null ? null : value.bytesDocument();
+         try {
+            var rootNode = mapper.readTree(doc);
+            var jpCtx = JsonPath.using(config).parse(rootNode);
+            // If no path provided return root
+            if (pathPos == arguments.size()) {
+               String resp = mapper.writer(rpp).writeValueAsString(rootNode);
+               return resp;
+            }
+            // If only 1 path provided return all the matching nodes as array
+            if (pathPos == arguments.size() - 1) {
+               var pathStr = new String(arguments.get(pathPos++));
+               JsonNode node = jpCtx.read(pathStr);
+               String resp = mapper.writer(rpp).writeValueAsString(node);
+               return resp;
+            }
+            // If more than 1 path provided return an object with
+            // properties "path": [array of matching nodes]
+            ObjectNode result = mapper.createObjectNode();
+            while (pathPos < arguments.size()) {
+               var pathStr = new String(arguments.get(pathPos++));
+               JsonNode node = jpCtx.read(pathStr);
+               result.set(pathStr, node);
+            }
+            String resp = mapper.writer(rpp).writeValueAsString(result);
+            return resp;
+         } catch (IOException e) {
+            return e.getMessage();
```

This is returning the exception as an OK status for RESP. Shouldn't we propagate the exception and return an exception status for the command?